processor/internal/transformer/destination_transformer/embedded/pubsub/pubsub.go

```diff
@@ -0,0 +1,161 @@
+package pubsub
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strings"
+
+	"github.com/rudderlabs/rudder-go-kit/stringify"
+	backendconfig "github.com/rudderlabs/rudder-server/backend-config"
+	utils "github.com/rudderlabs/rudder-server/processor/internal/transformer/destination_transformer/embedded"
+	types "github.com/rudderlabs/rudder-server/processor/types"
+	"github.com/rudderlabs/rudder-server/utils/misc"
+)
+
+var sourceKeys = []string{"properties", "traits", "context.traits"}
+
+func Transform(_ context.Context, events []types.TransformerEvent) types.Response {
```

Comment:
Deciding to implement `Transform` as function rather than a method makes sense with existing codebase. However, if in the feature we would like to add stats/logs or config access, a struct needs to be introduce to facilitate DI and avoid the use of globals. 

Let's keep simple for now.